backends/advanced/src/advanced_omi_backend/cron_scheduler.py

```
@@ -0,0 +1,277 @@
+            cfg.last_error = str(e)
+            logger.error(f"Cron job '{job_id}' failed: {e}", exc_info=True)
+            # Still advance next_run so we don't spin on failures
+            cfg.last_run = now
+            cfg.next_run = croniter(cfg.schedule, now).get_next(datetime)
+            await self._persist_state(job_id)
+            return {"error": str(e)}
+        finally:
+            cfg.running = False
+
+    async def _loop(self) -> None:
+        """Main scheduler loop – checks every 30s for due jobs."""
```

Ruff (0.14.14)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)